### PR TITLE
Fix LocalLayer Modifier(AvatarInventoryTradableItemRemover)

### PR DESCRIPTION
**Pull request by hskim881028**
### Description

SSIA

### How to test
Play
- EnhanceEquipment
- Combination
- Sell
- Sell cancel
- Buy
- Cook

### Related Links

https://github.com/planetarium/NineChronicles/issues/228

### Required Reviewers

@Namyujeong , @planetarium/9c-dev

### CODE REVIEW

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs [hunk 1]
```diff
@@ -278,15 +279,15 @@ private void ResponseUnrenderItemEnhancement(ActionBase.ActionEvaluation<ItemEnh
 
             // NOTE: 사용한 자원에 대한 레이어 다시 추가하기.
             LocalLayerModifier.ModifyAgentGold(agentAddress, -result.gold);
-            LocalLayerModifier.RemoveItem(avatarAddress, itemUsable.ItemId);
+            LocalLayerModifier.RemoveItem(avatarAddress, itemUsable.ItemId, itemUsable.RequiredBlockIndex, 1);
             foreach (var itemId in result.materialItemIdList)
             {
-                // NOTE: 최종적으로 UpdateCurrentAvatarState()를 호출한다면, 그곳에서 상태를 새로 설정할 것이다.
-                LocalLayerModifier.RemoveItem(avatarAddress, itemId);
+                avatarState.inventory.TryGetNonFungibleItem(itemId, out ItemUsable materialItem);
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:285]: Is `if(...)` check is not needed here?

File: nekoyume/Assets/_Scripts/State/LocalLayerModifier.cs [hunk 2]
```diff
@@ -687,16 +689,16 @@ Address slotAddress
...
                 .EnhancementCostSheet.Values
                 .FirstOrDefault(x => x.Grade == equipment.Grade && x.Level == equipment.level);
-            if (enhancementRow is null)
-            {
-                return;
-            }
-
-            equipment.LevelUp();
-            equipment.Update(requiredBlockIndex);
+            // if (enhancementRow is null)
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/State/LocalLayerModifier.cs:698]: This comment seems to be unnecessary now.

File: nekoyume/Assets/_Scripts/State/LocalLayerModifier.cs [hunk 3]
```diff
@@ -687,16 +689,16 @@ Address slotAddress
...
+            // {
+            //     return;
+            // }
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/State/LocalLayerModifier.cs:701]: This comment seems to be unnecessary now.

File: nekoyume/Assets/_Scripts/State/Modifiers/AvatarInventoryTradableItemRemover.cs [hunk 4]
```diff
@@ -1,48 +1,34 @@
...
-        public AvatarInventoryTradableItemRemover(Dictionary<Guid, int> dictionary)
-        {
-            innerDictionary = new InnerDictionary();
-            foreach (var pair in dictionary)
-            {
-                if (pair.Value is 0)
-                {
-                    continue;
-                }
+        private List<InnerModel> _items = new List<InnerModel>();
```
Comment (Unengine) [inline nekoyume/Assets/_Scripts/State/Modifiers/AvatarInventoryTradableItemRemover.cs:25]: How about mapping into dictionary like this?
```
private class InnerModel
{
            public long RequiredBlockIndex { get; }
            public int Count { get; set; }
}

private Dictionary<JsonConvertibleGuid, InnerModel> _items = new List<InnerModel>();
```

Comment (hskim881028) [reply nekoyume/Assets/_Scripts/State/Modifiers/AvatarInventoryTradableItemRemover.cs]: Please review one more time. :pray:

**Review by Unengine — APPROVED** [review]

**Pull request merged**